src/Quizzer.API.DAL.Tests/ApiDALTests.cs

```diff
@@ -0,0 +1,59 @@
+using ActiveTracker.DAL.Tests;
+using Microsoft.EntityFrameworkCore;
+using Quizzer.API.DAL.Entities;
+using Quizzer.API.DAL.Repository.Interfaces;
+using Quizzer.API.DAL.Seeds;
+using Quizzer.API.DAL.UnitOfWork;
+
+namespace Quizzer.API.DAL.Tests {
+
+    public class ApiDALTests : IAsyncLifetime {
+
+        protected UnitOfWorkFactory UnitOfWorkFactory;
+        protected IUnitOfWork UnitOfWork;
+
+        protected IDbContextFactory<QuizzerDbContext> DbContextFactory { get; }
+        protected QuizzerDbContext DbContextSUT { get; }
+
+        protected Storage storage;
+
+        public ApiDALTests() {
+            DbContextFactory = new QuizzerDbContextTestingFactory(GetType().FullName!, seedTestingData: true);
+            DbContextSUT = DbContextFactory.CreateDbContext();
+            UnitOfWorkFactory = new UnitOfWorkFactory(DbContextFactory);
+            UnitOfWork = UnitOfWorkFactory.Create();//initialize UnitOfWork so it doesn't have to be nullable
+            storage = new Storage(true);
+        }
+
+        public async Task DisposeAsync() {
+            await DbContextSUT.Database.EnsureDeletedAsync();
+            await DbContextSUT.DisposeAsync();
+            _ = UnitOfWork.DisposeAsync();
+        }
+
+        public async Task InitializeAsync() {
+            await DbContextSUT.Database.EnsureDeletedAsync();
+            await DbContextSUT.Database.EnsureCreatedAsync();
+            UnitOfWork = UnitOfWorkFactory.Create();
+        }
+
+        [Fact]
+        public async Task Update_QuizEntity_Updated() {
+            //Arrange
+            IRepository<QuizEntity> repo = UnitOfWork.GetRepository<QuizEntity>();
+            QuizEntity baseEntity = storage.Quizzes[0];
+            QuizEntity updatedEntity = baseEntity with { Name = baseEntity.Name + "Updated" };
+
+            //Act
+            Guid? repoEntityId = repo.Update(updatedEntity);
+            await UnitOfWork.CommitAsync();
+
+            //Assert
+            Assert.NotNull(repoEntityId);
+            Assert.Equal(repoEntityId, updatedEntity.Id);
+            await using QuizzerDbContext dbx = await DbContextFactory.CreateDbContextAsync();
+            QuizEntity dbxEntity = await dbx.Quiezzes.SingleAsync(i => i.Id == repoEntityId);
+            DeepAssert.Equal<QuizEntity>(updatedEntity, dbxEntity);
+        }
```

Comment:
I assume this is the test, which validates the saves changes inside the repository which was removed?

Reply:
yes